app/helpers/sellers_helper.rb

```diff
@@ -5,36 +5,114 @@ def self.generate_seller_json(seller:)
     locations = seller.locations
     seller = seller.as_json
     seller['locations'] = locations.as_json
-    seller['amount_raised'] = SellersHelper.calculate_amount_raised(
+    gift_card_amount = SellersHelper.calculate_gift_card_amount(
       seller_id: seller['id']
     )
+    donation_amount = SellersHelper.calculate_donation_amount(
+      seller_id: seller['id']
+    )
+    seller['gift_card_amount'] = gift_card_amount
+    seller['donation_amount'] = donation_amount
+    seller['amount_raised'] = gift_card_amount + donation_amount
+    num_gift_cards = SellersHelper.calculate_num_gift_cards(
+      seller_id: seller['id']
+    )
+    num_donations = SellersHelper.calculate_num_donations(
+      seller_id: seller['id']
+    )
+    seller['num_gift_cards'] = num_gift_cards
+    seller['num_donations'] = num_donations
+    seller['num_contributions'] = num_gift_cards + num_donations
+
     seller
   end
 
-  # Calculates the amount of money raised by the Seller so far.
+  # calculates the amount raised from gift cards
   # seller_id: the actual id of the Seller. Seller.id
-  def self.calculate_amount_raised(seller_id:)
+  def self.calculate_gift_card_amount(seller_id:)
     return 0 if Item.where(seller_id: seller_id).empty?
 
+    gift_card_amount = GiftCardDetail.joins(:item)
+                                     .where(items: {
+                                              seller_id: seller_id,
+                                              refunded: false
+                                            })
+                                     .inject(0) do |sum, gift_card| sum + gift_card.amount
+                                      end
+    gift_card_amount
+    end
+
+  # calculates the amount raised from donations
+  # seller_id: the actual id of the Seller. Seller.id
+  def self.calculate_donation_amount(seller_id:)
     donation_amount = DonationDetail.joins(:item)
                                     .where(items: {
                                              seller_id: seller_id,
                                              refunded: false
                                            })
-                                    .inject(0) do |sum, donation|
-      sum + donation.amount
+                                    .inject(0) do |sum, donation| sum + donation.amount
```

Comment:
`sum + donation.amount` should be in its own line

but just run rubocop -a and it'll fix it for you